pymc_marketing/mmm/tvp.py

```diff
@@ -0,0 +1,73 @@
+from typing import Optional
+
+import pymc as pm
+from pymc_marketing.mmm.utils import softplus
+
+
+def time_varying_prior(
+    name: str,
+    X: pm.Deterministic,
+    X_mid: int | float,
+    positive: bool = False,
+    dims: Optional[tuple[str, str] | str] = None,
+    m: int = 40,
+    L: int = 100,
+    eta_lam: float = 1,
+    ls_mu: float = 5,
+    ls_sigma: float = 5,
+    cov_func: Optional[pm.gp.cov.Prod] = None,
+    model: Optional[pm.Model] = None,
+) -> pm.Deterministic:
+    """Time varying prior, based the Hilbert Space Gaussian Process (HSGP).
+
+    Parameters
+    ----------
+    name : str
+        Name of the prior.
+    X : 1d array-like of int or float
+        Time points.
+    X_mid : int or float
+        Midpoint of the time points.
+    positive : bool
+        Whether the prior should be positive.
+    dims : tuple of str or str
+        Dimensions of the prior.
+    m : int
+        Number of basis functions.
+    L : int
+        Number of quadrature points.
+    eta_lam : float
+        Exponential prior for the variance.
+    ls_mu : float
+        Mean of the inverse gamma prior for the lengthscale.
+    ls_sigma : float
+        Standard deviation of the inverse gamma prior for the lengthscale.
+    cov_func : pm.gp.cov.Prod
+        Covariance function.
+    model : pm.Model
+        PyMC model.
+
+    Returns
+    -------
+    pm.Deterministic
+        Time-varying prior.
+    """  # noqa: W605
+
+    with pm.modelcontext(model) as model:
+        if cov_func is None:
+            eta = pm.Exponential(f"eta_{name}", lam=eta_lam)
```

Comment:
I think using this current implementation, we could move these distributions into the `model_config`?

Reply:
oh this would be really clever, it just becomes another supported distribution?

Reply:
that's better than an hsgp_config I think. I'm trying to work out what's the right level of rigidity to build into this thing. There are some things here you wouldn't want to change as a user I think. And maybe this is just me, but as we put distributions into the config, we pay the price of more obscure code. Opinions?